src/ui/win32/Dlg_MessageBox.cpp

```diff
@@ -0,0 +1,121 @@
+#include "Dlg_MessageBox.hh"
+
+#include "RA_Defs.h" // for NativeStr
+
+#include "ui/viewmodels/MessageBoxViewModel.hh"
+
+#include <WTypes.h>
+
+#undef GetMessage // ignore Windows macro
+
+namespace ra {
+namespace ui {
+namespace win32 {
+
+using ra::ui::viewmodels::MessageBoxIcon;
+using ra::ui::viewmodels::MessageBoxButtons;
+using ra::ui::viewmodels::MessageBoxViewModel;
+
+static HRESULT(WINAPI *pTaskDialog)(HWND hwndParent, HINSTANCE hInstance, PCWSTR pszWindowTitle, PCWSTR pszMainInstruction,
```

Comment:
I strongly suggest using `std::function` for this callback, seems kind of strange making a global static but not too a big deal. It would look something like this. Adjust it however you like, it's just an example.
```
using TaskDialogCallbackType = HRESULT(WINAPI*)(HWND, HINSTANCE, PCWSTR, PCWSTR, TASKDIALOG_COMMON_BUTTON_FLAGS, PCWSTR, int);
std::function<TaskDialogCallbackType > TaskDialogCallback;
```

`TASKDIALOG_COMMON_BUTTON_FLAGS `is Vista+ so I'd suggest surround this callback with
```
#ifndef _WIN32_WINNT >= _WIN32_WINNT_LONGHORN
...
#endif // _WIN32_WINNT >= _WIN32_WINNT_LONGHORN
```
if supporting Windows XP is still considered a requirement. The "dynamically" finding doesn't really matter because the SDK is for Windows 7, the last platform SDK to support XP.